Source/Core/Core/PowerPC/Jit64/Jit_Integer.cpp

```diff
@@ -1618,34 +1619,34 @@ void Jit64::rlwnmx(UGeckoInstruction inst)
   INSTRUCTION_START
   JITDISABLE(bJITIntegerOff);
   int a = inst.RA, b = inst.RB, s = inst.RS;
+  auto ra = regs.gpr.Lock(a);
+  auto rb = regs.gpr.Lock(b);
+  auto rs = regs.gpr.Lock(s);
 
   u32 mask = Helper_Mask(inst.MB, inst.ME);
-  if (gpr.R(b).IsImm() && gpr.R(s).IsImm())
+  if (rb.IsImm() && rs.IsImm())
   {
-    gpr.SetImmediate32(a, _rotl(gpr.R(s).Imm32(), gpr.R(b).Imm32() & 0x1F) & mask);
+    ra.SetImm32(_rotl(rs.Imm32(), rb.Imm32() & 0x1F) & mask);
   }
   else
   {
     // no register choice
-    gpr.FlushLockX(ECX);
-    gpr.Lock(a, b, s);
-    MOV(32, R(ECX), gpr.R(b));
-    gpr.BindToRegister(a, (a == s), true);
+    auto ecx = regs.gpr.Borrow(ECX);
```

Comment:
Oh, okay, that makes sense; I only skimmed over the regcache interface, and there weren't any comments about that sort of thing.